### PR TITLE
fix: Handling broken token issues

### LINKED ISSUE #1293: Invalid auth tokens fail confusingly
**Describe the bug**
When the saved auth token (`$HOME/.flank/UserToken`) is invalid for any reason, it is hard to understand why it is invalid. It also does not clean itself up.

**To Reproduce**
> Steps to reproduce the behavior:

1. Run `java -jar flank.jar auth login`
2. Login
3. Wait long enough for this token to expire (I don't know how long this is)
4. Run flank again on the same computer

**Expected behavior**
> A clear and concise description of what you expected to happen.

I don't have a clear idea of what expected behavior is.
Idea 1 - Detect that the token is invalid, throw an error, and tell the user to run `auth login` again.
Idea 2- Automatically delete the token and fall back to env var credentials (via `$GOOGLE_APPLICATION_CREDENTIALS`)
Idea 3 - Clear log statement showing where flank is trying to pull credentials from, so at least when they see auth related errors, they know where to look

**Details (please complete the following information):**
> Have you tested on the latest Flank snapshot?
>
> Post the output of ` flank --version`.
Actual behavior:

I believe I had an expired token. When I tried running flank, I saw the error:
```
Caused by: com.google.api.client.http.HttpResponseException: 400 Bad Request
POST https://oauth2.googleapis.com/token
{
  "error": "invalid_grant",
  "error_description": "Bad Request"
}
```

I was playing around with `$GOOGLE_APPLICATION_CREDENTIALS`, but I had to manually step through the code to realize that I had a locally saved token that was taking precedence over the env var.

**Additional context**
> Add any other context about the problem here.

### PR DESCRIPTION
**Pull request by piotradamczyk5**
Fixes #1293

When there is a problem with the token listed in the checklist, a proper message with fix instructions is displayed and a bad token file is deleted

## Checklist

- [x] Handle expired token
- [x] Handle revoked token
- [x] Handle issues during serialization
- [x] Unit tested

### CODE REVIEW

**Comment by Sloox** [comment]
does this fix the windows issues with google tokens? 👍

**Comment by piotradamczyk5** [comment]
> does this fix the windows issues with google tokens? 👍

I think that it was fixed before by @adamfilipow92 
This fixes expired or bad token handling

File: test_runner/src/test/kotlin/ftl/gc/GcToolResultsTest.kt [hunk 1]
```diff
@@ -149,9 +156,40 @@ class GcToolResultsTest {
...
+            )
             val exception = getThrowable { GcToolResults.getDefaultBucket(projectId) }
             assertEquals(expected, exception.message)
         }
     }
+
+    @Test
+    fun `getDefaultBucket on 400 with authentication error should throw exception with specific message`() {
+        val expected = """
+Could not load user authentication, please
```
Comment (Sloox) [inline test_runner/src/test/kotlin/ftl/gc/GcToolResultsTest.kt:170]: strange line wrapping.

**Review by Sloox — APPROVED** [review]

**Comment by piotradamczyk5** [comment]
@Mergifyio rebase

**Review by adamfilipow92 — APPROVED** [review]

**Pull request merged**